test/layers.jl

```diff
@@ -0,0 +1,228 @@
+
+    act!([l1, l2, l3], layer_actions)
+
+    ball1 = Object(5:41, (args...) -> object_layer(O, "red"), Point(50, 0))
+    act!(ball1, Action(anim_rotate_around(2π, O)))
+    ball2 = Object(5:41, (args...) -> object_layer(O, "blue"), Point(100, 40))
+    act!(ball2, Action(anim_rotate_around(2π, 0.0, ball1)))
+    act!(ball2, Action(10:30, disappear(:fade)))
+    conn = Object(5:41, (args...) -> connector(pos(ball1), pos(ball2), "black"))
+    por = Object(5:41, (args...) -> path!(path_of_red, pos(ball1), "red"))
+    pob = Object(5:41, (args...) -> path!(path_of_blue, pos(ball2), "blue"))
+    layer_objects = [ball1, ball2, conn, por, pob]
+
+    @testset "Layer macro" begin
+        @test l1.frames.frames == l2.frames.frames == l3.frames.frames
+        @test l1.width == l2.width == l3.width == 600
+        @test l1.height == l2.height == l3.height == 600
+        @test l1.position == l2.position == l3.position == Point(0, 0)
+        @test length(l1.layer_objects) ==
+              length(l2.layer_objects) ==
+              length(l3.layer_objects) ==
+              length(layer_objects) + 2
+        @test length(l1.actions) ==
+              length(l2.actions) ==
+              length(l3.actions) ==
+              length(layer_actions)
+        @test l1.current_setting.opacity ==
+              l2.current_setting.opacity ==
+              l3.current_setting.opacity ==
+              1.0
+        @test l1.current_setting.scale ==
+              l2.current_setting.scale ==
+              l3.current_setting.scale ==
+              Javis.Scale(1.0, 1.0)
+        @test l1.current_setting.rotation_angle ==
+              l2.current_setting.rotation_angle ==
+              l3.current_setting.rotation_angle ==
+              0.0
+        @test l1.image_matrix == l2.image_matrix == l3.image_matrix == nothing
+    end
+
+    # remove duplicate layers after above testing
+    video.layers = [l1]
+    @test get_position(l1) == Point(0, 0)
+
+    astar(args...; do_action = :stroke) = star(O, 50, 5, 0.5, 0, do_action)
+    acirc(args...; do_action = :stroke) = circle(Point(100, 100), 50, do_action)
+    star_obj = Object(1:30, astar)
+    act!(star_obj, Action(linear(), morph_to(acirc)))
+    circle_obj = Object(31:60, acirc)
+    act!(circle_obj, Action(linear(), morph_to(astar)))
+
+    opacity_anim = Animation(
+        [0, 0.5, 1], # must go from 0 to 1
+        [0.0, 0.3, 0.7],
+        [sineio(), sineio()],
+    )
+
+    l4 = @JLayer 5:20 200 200 Point(-50, 50) :transparent begin
+        Background(1:14, ground2)
+        rball = Object(1:14, (args...) -> object_layer(O, "black"), Point(0, 0))
+        act!(rball, Action(anim_translate(Point(20, -20))))
+    end
+
+    act!(l4, Action(1:5, opacity_anim, setopacity()))
+
+    Javis.show_layer_frame(71:79, 5:12, l1)
+    Javis.show_layer_frame(71:79, 10, l4)
+
+
+    render(video; tempdirectory = "images", pathname = "layer_test.gif")
+
+    for i in [
+        "03",
+        "05",
+        "08",
```

Comment:
Why are those strings? Maybe use `lpad` in line 184 instead

Reply:
since the file name is `images/00000000$i.png`, there's an extra 0 for single digit numbers. Hence the strings. eg I get an error:
`ArgumentError: No file exists at given path: 0000000003.png` without the string

Reply:
I'm not sure what you mean with your last sentence. `lpad` is made for this kind of problem (well you can also use it for others)

Reply:
ah got it! thanks